File: arrays/single_number/single_number.go
package main

func findSingleNumber(value int, index int, nums []int, f chan bool, n chan int) {
    for ind, val := range nums {
        if index == ind {
            continue
        } else if value == val {
            f <- false
            return
        }
    }
    f <- true
    n <- value
}


func singleNumber(nums []int) int {
    f := make(chan bool)
    n := make(chan int)
    singlenumber := 0

    for index, value := range nums {
        go findSingleNumber(value, index, nums, f, n)

        found := <-f
        if found {
            singlenumber = <-n
            break
        }
    }

    return singlenumber
}

File: arrays/single_number/single_number.py
def singleNumber(nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    length = len(nums)
    found = False
    singlevalue = 0
    for i, ivalue in enumerate(nums):
        if found:
            break
        else:
            currindex = i
            singlevalue = ivalue
            for j, jvalue in enumerate(nums):
                if (j == currindex):
                    continue
                elif (jvalue == singlevalue):
                    break
                elif (j == length-1):
                    found = True
                    break
    return singlevalue

#print singleNumber([1])
print singleNumber([1, 0, 1])

